src/generator/AutoRest.Go/Model/MethodGo.cs

```diff
@@ -213,6 +214,10 @@ public string MethodReturnSignature(bool helper)
             return $"{retVal}, {errVal}";
         }
 
+        public bool ShouldValidate() {
+            return !ParameterValidations.IsNullOrEmpty();
```

Comment:
But will ParameterGo.Validate() actually get called when ShouldValidate() == false?  I would think it not to get called.